src/tools/clippy/tests/ui/pattern_type_mismatch/syntax.stderr

```diff
@@ -23,14 +23,6 @@ LL |     while let Some(_) = ref_value {
    |
    = help: use `*` to dereference the match expression or explicitly match against a `&_` pattern and adjust the enclosed variable bindings
 
-error: type of pattern does not match the expression type
```

Comment:
Any clue why this one got lost? If not, @rust-lang/clippy should we track this in an issue to get it reinstated?

Reply:
I went ahead and fixed this.